mentat/terminal/terminal_app.py

```diff
@@ -0,0 +1,307 @@
+        if self.loading_bar is not None:
+            self.loading_bar.remove()
+
+
+class ContextContainer(Static):
+    def _build_path_tree(self, files: list[str], cwd: Path):
+        """Builds a tree of paths from a list of CodeFiles."""
+        tree = dict[str, Any]()
+        for file in files:
+            path = os.path.relpath(file, cwd)
+            parts = Path(path).parts
+            current_level = tree
+            for part in parts:
+                if part not in current_level:
+                    current_level[part] = {}
+                current_level = current_level[part]
+        return tree
+
+    def _build_sub_tree(
+        self,
+        cur_path: Path,
+        root: TreeNode[Any],
+        children: Dict[str, Any],
+        git_diff_paths: Set[Path],
+    ):
+        for child, grandchildren in children.items():
+            new_path = cur_path / child
+            if not grandchildren:
+                if new_path in git_diff_paths:
+                    label = f"[green]* {child}[/green]"
+                else:
+                    label = child
+                root.add_leaf(label)
+            else:
+                child_node = root.add(child, expand=True)
+                self._build_sub_tree(
+                    new_path, child_node, grandchildren, git_diff_paths
+                )
+
+    def _build_tree_widget(
+        self, files: list[str], cwd: Path, git_diff_paths: Set[Path]
+    ) -> Tree[Any]:
+        path_tree = self._build_path_tree(files, cwd)
+        tree: Tree[Any] = Tree(f"[blue]{cwd.name}[/blue]")
+        tree.root.expand()
+        self._build_sub_tree(cwd, tree.root, path_tree, git_diff_paths)
+        return tree
+
+    def update_context(
+        self,
+        cwd: Path,
+        diff_context_display: Optional[str],
+        auto_context_tokens: int,
+        features: List[str],
+        auto_features: List[str],
+        git_diff_paths: Set[Path],
+        total_tokens: int,
+        total_cost: float,
+    ):
+        feature_tree = self._build_tree_widget(features, cwd, git_diff_paths)
+        auto_feature_tree = self._build_tree_widget(auto_features, cwd, git_diff_paths)
+
+        context_header = ""
+        context_header += "[blue bold]Code Context:[/blue bold]"
+        context_header += f"\nTokens: [yellow]{total_tokens}[/yellow]"
+        context_header += f"\nTotal Session Cost: [yellow]${total_cost:.2f}[/yellow]"
+        context_header += f"\nDirectory: {cwd}"
+        if diff_context_display:
+            context_header += f"\nDiff:[green]{diff_context_display}[/green]"
+        if auto_context_tokens > 0:
+            context_header += (
+                f"\nAuto-Context: Enabled\nAuto-Context Tokens: {auto_context_tokens}"
+            )
+        else:
+            context_header += "\nAuto-Context: [yellow]Disabled[/yellow]"
+
+        context_header += "\nIncluded Files:"
+        if not features:
+            context_header += " [yellow]None[/yellow]"
```

Comment:
Can all of these inline colors/styles be send as `extra` items in the session stream? Would be easier to parse for the VSCode extension (and for any other extensions building ontop of Mentat)